from PyQt5.uic import loadUi
from PyQt5.QtWidgets import QDialog
from views.misc.messagebox import MessageBox


class NewProfileView(QDialog):
    def __init__(self, profilesController):
        super(NewProfileView, self).__init__()
        self.__profilesController = profilesController
        loadUi('./resources/ui/dialog/newprofile.ui', self)

        # Connect things
        self.buttonBox.rejected.connect(self.buttonBoxRejected)
        self.buttonBox.accepted.connect(self.buttonBoxAccepted)

        self.__profiles = None

    def buttonBoxRejected(self):
        self.reject()

    def buttonBoxAccepted(self):
        name = self.lineEditProfile.text()
        command = self.lineEditCommand.text()
        if name:
            if command:
                self.__profiles.setProfile(name, command)
                self.done(1)
            else:
                MessageBox.critical("Error", "Command box is empty")
                self.done(-1)
        else:
            MessageBox.critical("Error", "Profile's name is empty")
            self.done(-2)

    def setProfiles(self, profiles):
        self.__profiles = profiles
